File: 2017/1.py


data = list(map(int, list(open("1.txt").read()[:-1])))
sum = 0
n = len(data)
for i in range(1, n):
    if data[i] == data[i-1]:
        sum += data[i]

if data[0] == data[-1]:
    sum += data[0]

print("Part1:", sum)

sum = 0
for i in range(len(data)):
    if data[i] == data[(i+n//2)%n]:
        sum += data[i]

print("Part2:", sum)

File: 2015 (Go)/1.go
package main

import (
	"fmt"
	"io/ioutil"
)

func main() {
	data, err := ioutil.ReadFile("1.txt")
	if err != nil {
		fmt.Println("Error while reading file!", err)
		return
	}
	var count = 0
	for _, c := range string(data) {
		if string(c) == "(" {
			count++
		} else {
			count--
		}
	}
	fmt.Println("Part1:", count)

	count = 0
	for i, c := range string(data) {
		if string(c) == "(" {
			count++
		} else {
			count--
		}
		if count == -1 {
			fmt.Println("Part2:", i+1)
			break
		}
	}

}
